include/oneapi/dpl/pstl/hetero/algorithm_ranges_impl_hetero.h

```diff
@@ -391,15 +472,29 @@ __pattern_search(__hetero_tag<_BackendTag> __tag, _ExecutionPolicy&& __exec, _Ra
 }
 
 #if _ONEDPL_CPP20_RANGES_PRESENT
+template <typename _Pred, typename _Proj1, typename _Proj2>
+struct __pattern_search_pred
```

Comment:
It seems, we already have a type with identical functionality. For example,
`struct __pattern_equal_pred`, see above.
I believe we can use one such type in the all places where such functor is called.

Reply:
Fixed.